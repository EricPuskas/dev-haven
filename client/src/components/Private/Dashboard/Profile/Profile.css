.ProfileNew {
  height: 100vh;
  background-color: rgb(250, 250, 250);
  text-align: center;
}

.ProfileNew h1 {
  color: #4f4f4f;
}

.ProfileNew p {
  color: #4f4f4f;
  font-size: 1.2rem;
}

.ProfileNew small {
  color: #4f4f4f;
  font-size: 0.9rem;
}

.ProfileNew .form-control {
  background-color: #fafafa;
}
.ProfileNew .btn.btn-standard {
  border: 1px solid #e54f4f;
  color: #e54f4f;
  padding: 0.2rem 1rem;
  font-size: 1rem;
}

.ProfileNew .btn.btn-standard.btn-submit {
  padding: 0.1rem 3rem;
  font-size: 1.2rem;
  background-color: #e54f4f;
  color: #fff;
}
.ProfileNew .ProfileNew-optional {
  font-size: 0.8rem;
  color: #4f4f4f;
  margin: 0 1rem;
}

.ProfileNew-social-inputs {
  margin: 2rem 0;
}

.ProfileNew-container {
  display: flex;
  justify-content: center;
  flex-direction: column;
  background-color: rgb(250, 250, 250);
  width: 100%;
  align-items: center;
  text-align: start;
  height: auto;
  padding: 2rem;
  margin: 0 auto;
  margin-bottom: 5rem;
}

.ProfileNew-container .main-div {
  max-width: 800px;
}

.form-text {
  display: block;
  margin-top: 0;
  margin-bottom: 1rem;
}

.form-group {
  margin-bottom: 0rem;
  margin-top: 1rem;
}

textarea {
  width: 100%;
  padding: 0.5rem;
  background-color: #fafafa;
}
.ProfileNew-controls {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  align-items: center;
  margin-top: 2rem;
}

.btn.btn-standard.btn-go-back {
  border: 1px solid rgb(117, 117, 117);
  color: rgb(117, 117, 117);
}

.errorMsg {
  font-size: 0.7rem;
  margin-top: 0.4rem;
  color: #dc3545;
  text-align: end;
  display: block;
  float: right;
}
