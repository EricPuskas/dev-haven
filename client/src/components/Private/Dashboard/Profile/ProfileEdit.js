import React, { Fragment, useState, useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { createProfile } from "../../../../actions/profile";
import { getCurrentProfile } from "../../../../actions/auth";
import { Link, withRouter } from "react-router-dom";
import Input from "../../../Common/Input";
import Select from "../../../Common/Select";
import statusArray from "../../../../utils/status";

const ProfileEdit = ({
  auth: { user },
  profile: { loading },
  createProfile,
  getCurrentProfile,
  history,
  errors
}) => {
  const [formData, setFormData] = useState({
    company: "",
    website: "",
    location: "",
    status: "",
    skills: "",
    githubusername: "",
    bio: "",
    twitter: "",
    facebook: "",
    linkedin: "",
    youtube: "",
    instagram: ""
  });
  const [displaySocialInputs, toggleSocialInputs] = useState(false);

  const {
    company,
    website,
    location,
    status,
    skills,
    githubusername,
    bio,
    twitter,
    facebook,
    linkedin,
    youtube,
    instagram
  } = formData;

  const { error } = errors;
  useEffect(() => {
    getCurrentProfile();
    setFormData({
      company: loading || !user.company ? "" : user.company,
      website: loading || !user.website ? "" : user.website,
      location: loading || !user.location ? "" : user.location,
      status: loading || !user.status ? "" : user.status,
      skills: loading || !user.skills ? "" : user.skills.join(","),
      githubusername:
        loading || !user.githubusername ? "" : user.githubusername,
      bio: loading || !user.bio ? "" : user.bio,
      twitter: loading || !user.social ? "" : user.social.twitter,
      facebook: loading || !user.social ? "" : user.social.facebook,
      linkedin: loading || !user.social ? "" : user.social.linkedin,
      youtube: loading || !user.social ? "" : user.social.youtube,
      instagram: loading || !user.social ? "" : user.social.instagram
    }); // eslint-disable-next-line
  }, [getCurrentProfile]);

  const handleChange = e =>
    setFormData({
      ...formData,
      [e.target.name]: e.target.value
    });
  const handleSubmit = e => {
    console.log(formData);
    e.preventDefault();
    createProfile(formData, history, true);
  };
  return (
    <div className="ProfileNew">
      <h1> Create your profile!</h1>
      <p> Let's get some information to make your profile stand out </p>
      <small>* = required field</small>
      <div className="ProfileNew-container">
        <div className="main-div">
          <form className="ProfileNew-form" onSubmit={handleSubmit}>
            <Select
              name="status"
              onChange={handleChange}
              icon="fas fa-building"
              defaultValue={user.status}
              options={statusArray}
              error={error.status}
              error_fixed
            />
            <small className="form-text">
              Give us an idea of where you are at in your career
            </small>
            <Input
              placeholder="Company"
              name="company"
              icon="fas fa-building"
              type="text"
              value={company}
              onChange={handleChange}
              error={error.company}
              error_fixed
              mb={0}
            />
            <small className="form-text">
              Could be your own company or one you work for
            </small>
            <Input
              placeholder="Location"
              name="location"
              icon="fas fa-map-marker-alt"
              type="text"
              value={location}
              onChange={handleChange}
              error={error.location}
              error_fixed
              mb={0}
            />
            <small className="form-text">
              City & state suggested (eg. Boston, MA)
            </small>
            <Input
              placeholder="* Skills"
              name="skills"
              icon="far fa-check-circle"
              type="text"
              value={skills}
              onChange={handleChange}
              error={error.skills}
              error_fixed
              mb={0}
            />
            <small className="form-text">
              Please use comma separated values (eg. HTML,CSS,JavaScript,PHP)
            </small>
            <Input
              placeholder="Github Username"
              name="githubusername"
              icon="fab fa-github"
              type="text"
              value={githubusername}
              onChange={handleChange}
              error={error.githubusername}
              error_fixed
              mb={0}
            />
            <small className="form-text">
              If you want your latest repos and a Github link, include your
              username
            </small>
            <div className="form-group">
              <textarea
                placeholder="A short bio of yourself"
                value={bio}
                maxLength={500}
                rows={5}
                onChange={handleChange}
                name="bio"
                style={{ width: "100%" }}
              />
              <small className="form-text">
                Tell us a little about yourself{" "}
                {`( ${500 - bio.length} characters left. )`}
              </small>
            </div>
            <div className="my-2">
              <button
                onClick={() => toggleSocialInputs(!displaySocialInputs)}
                type="button"
                className="btn btn-standard"
              >
                Add Social Network Links
              </button>
              <span className="ProfileNew-optional">Optional</span>
            </div>
            {displaySocialInputs && (
              <div className="ProfileNew-social-inputs">
                <Input
                  placeholder="Twitter URL"
                  name="twitter"
                  icon="fab fa-twitter"
                  type="text"
                  value={twitter}
                  onChange={handleChange}
                  error={error.twitter}
                  error_fixed
                />
                <Input
                  placeholder="Facebook URL"
                  name="facebook"
                  icon="fab fa-facebook"
                  type="text"
                  value={facebook}
                  onChange={handleChange}
                  error={error.facebook}
                  error_fixed
                />
                <Input
                  placeholder="Youtube URL"
                  name="youtube"
                  icon="fab fa-youtube"
                  type="text"
                  value={youtube}
                  onChange={handleChange}
                  error={error.youtube}
                  error_fixed
                />
                <Input
                  placeholder="Linkedin URL"
                  name="linkedin"
                  icon="fab fa-linkedin"
                  type="text"
                  value={linkedin}
                  onChange={handleChange}
                  error={error.linkedin}
                  error_fixed
                />
                <Input
                  placeholder="Instagram URL"
                  name="instagram"
                  icon="fab fa-instagram"
                  type="text"
                  value={instagram}
                  onChange={handleChange}
                  error={error.instagram}
                  error_fixed
                />
              </div>
            )}
            <div className="ProfileNew-controls">
              <button
                type="submit"
                className="btn btn-standard btn-submit"
                id="ProfileNew-submit-btn"
              >
                {loading ? "Loading..." : "Save"}
              </button>
              <Link className="btn btn-standard btn-go-back" to="/dashboard">
                Go Back
              </Link>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  auth: state.auth,
  profile: state.profile,
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { createProfile, getCurrentProfile }
)(withRouter(ProfileEdit));
