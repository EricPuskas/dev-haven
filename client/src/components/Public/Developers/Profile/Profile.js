import React, { Fragment, useEffect } from "react";
import { connect } from "react-redux";
import Loader from "../../../../assets/img/rolling.svg";
import { getProfileByID } from "../../../../actions/profile";
import { Link } from "react-router-dom";
import ProfileTop from "./ProfileTop";
import ProfileAbout from "./ProfileAbout";
import ProfileExperience from "./ProfileExperience";
import ProfileEducation from "./ProfileEducation";
import ProfileGithub from "./ProfileGithub";

const Profile = ({
  match,
  getProfileByID,
  profile: { profile, loading },
  auth
}) => {
  useEffect(() => {
    getProfileByID(match.params.id);
  }, [getProfileByID, match.params.id]);
  return (
    <div className="Developer-profile">
      {profile === null || loading ? (
        <div className="Developers-loader-container" style={{ marginTop: 0 }}>
          <img src={Loader} alt="Loading..." className="Developers-loader" />
        </div>
      ) : (
        <>
          <Link to="/profiles" className="btn btn-standard btn-go-back">
            {" "}
            Go Back
          </Link>
          {auth.isAuthenticated &&
            auth.loading === false &&
            auth.user._id === profile.user._id && (
              <Link to="/edit-profile" className="btn btn-dark">
                Edit Profile
              </Link>
            )}
          <div className="profile-grid my-1">
            <ProfileTop profile={profile} />
            <ProfileAbout profile={profile} />
            <div className="profile-exp bg-white p-2">
              <h2 className="text-primary">Experience</h2>
              {profile.experience.length > 0 ? (
                <Fragment>
                  {profile.experience.map(exp => (
                    <ProfileExperience key={exp._id} experience={exp} />
                  ))}
                </Fragment>
              ) : (
                <h4>No experience credentials</h4>
              )}
            </div>
            <div className="profile-edu bg-white p-2">
              <h2 className="text-primary">Education</h2>
              {profile.education.length > 0 ? (
                <Fragment>
                  {profile.education.map(edu => (
                    <ProfileEducation key={edu._id} education={edu} />
                  ))}
                </Fragment>
              ) : (
                <h4>No education credentials</h4>
              )}
            </div>
            {profile.githubusername && (
              <ProfileGithub user={profile.githubusername} />
            )}
          </div>
        </>
      )}
    </div>
  );
};
const mapStateToProps = state => ({
  profile: state.profile,
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { getProfileByID }
)(Profile);
