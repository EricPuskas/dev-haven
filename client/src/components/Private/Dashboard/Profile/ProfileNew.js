import React, { useState } from "react";
import { connect } from "react-redux";
import { createProfile } from "../../../../actions/profile";
import { Link, withRouter } from "react-router-dom";
import Input from "../../../Common/Input";
import Select from "../../../Common/Select";
import statusArray from "../../../../utils/status";
import "./Profile.css";

const ProfileNew = ({ errors, createProfile, history }) => {
  const { error } = errors;
  const [formData, setFormData] = useState({
    company: "",
    website: "",
    location: "",
    status: "",
    skills: "",
    githubusername: "",
    bio: "",
    twitter: "",
    facebook: "",
    linkedin: "",
    youtube: "",
    instagram: ""
  });
  const [displaySocialInputs, toggleSocialInputs] = useState(false);
  const {
    company,
    website,
    location,
    status,
    skills,
    githubusername,
    bio,
    twitter,
    facebook,
    linkedin,
    youtube,
    instagram
  } = formData;

  const handleChange = e =>
    setFormData({
      ...formData,
      [e.target.name]: e.target.value
    });

  const handleSubmit = e => {
    e.preventDefault();
    createProfile(formData, history);
  };

  return (
    <div className="ProfileNew">
      <h1> Create your profile!</h1>
      <p> Let's get some information to make your profile stand out </p>
      <small>* = required field</small>
      <div className="ProfileNew-container">
        <div className="main-div">
          <form className="ProfileNew-form" onSubmit={handleSubmit}>
            <Select
              name="status"
              onChange={handleChange}
              profileFormGroup={true}
              icon="fas fa-building"
              defaultValue={"* Select Professional Status"}
              options={[statusArray]}
              error={error.status}
              error_fixed
            />
            <small className="form-text">
              Give us an idea of where you are at in your career
            </small>
            <Input
              placeholder="Company"
              name="company"
              icon="fas fa-building"
              type="text"
              value={company}
              onChange={handleChange}
              error={error.company}
              error_fixed
              mb={0}
            />
            <small className="form-text">
              Could be your own company or one you work for
            </small>
            <Input
              placeholder="Location"
              name="location"
              icon="fas fa-map-marker-alt"
              type="text"
              value={location}
              onChange={handleChange}
              error={error.location}
              error_fixed
              mb={0}
            />
            <small className="form-text">
              City & state suggested (eg. Boston, MA)
            </small>
            <Input
              placeholder="* Skills"
              name="skills"
              icon="far fa-check-circle"
              type="text"
              value={skills}
              onChange={handleChange}
              error={error.skills}
              error_fixed
              mb={0}
            />
            <small className="form-text">
              Please use comma separated values (eg. HTML,CSS,JavaScript,PHP)
            </small>
            <Input
              placeholder="Github Username"
              name="githubusername"
              icon="fab fa-github"
              type="text"
              value={githubusername}
              onChange={handleChange}
              error={error.githubusername}
              error_fixed
              mb={0}
            />
            <small className="form-text">
              If you want your latest repos and a Github link, include your
              username
            </small>
            <div className="form-group">
              <textarea
                placeholder="A short bio of yourself"
                value={bio}
                maxLength={500}
                rows={5}
                onChange={handleChange}
                name="bio"
                style={{ width: "100%" }}
              />
              <small className="form-text">
                Tell us a little about yourself{" "}
                {`( ${500 - bio.length} characters left. )`}
              </small>
            </div>
            <div className="my-2">
              <button
                onClick={() => toggleSocialInputs(!displaySocialInputs)}
                type="button"
                className="btn btn-standard"
              >
                Add Social Network Links
              </button>
              <span className="ProfileNew-optional">Optional</span>
            </div>
            {displaySocialInputs && (
              <div className="ProfileNew-social-inputs">
                <Input
                  placeholder="Twitter URL"
                  name="twitter"
                  icon="fab fa-twitter"
                  type="text"
                  value={twitter}
                  onChange={handleChange}
                  error={error.twitter}
                  error_fixed
                />
                <Input
                  placeholder="Facebook URL"
                  name="facebook"
                  icon="fab fa-facebook"
                  type="text"
                  value={facebook}
                  onChange={handleChange}
                  error={error.facebook}
                  error_fixed
                />
                <Input
                  placeholder="Youtube URL"
                  name="youtube"
                  icon="fab fa-youtube"
                  type="text"
                  value={youtube}
                  onChange={handleChange}
                  error={error.youtube}
                  error_fixed
                />
                <Input
                  placeholder="Linkedin URL"
                  name="linkedin"
                  icon="fab fa-linkedin"
                  type="text"
                  value={linkedin}
                  onChange={handleChange}
                  error={error.linkedin}
                  error_fixed
                />
                <Input
                  placeholder="Instagram URL"
                  name="instagram"
                  icon="fab fa-instagram"
                  type="text"
                  value={instagram}
                  onChange={handleChange}
                  error={error.instagram}
                  error_fixed
                />
              </div>
            )}
            <div className="ProfileNew-controls">
              <button
                type="submit"
                className="btn btn-standard btn-submit"
                id="ProfileNew-submit-btn"
              >
                Submit
              </button>
              <Link className="btn btn-standard btn-go-back" to="/dashboard">
                Go Back
              </Link>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { createProfile }
)(withRouter(ProfileNew));
