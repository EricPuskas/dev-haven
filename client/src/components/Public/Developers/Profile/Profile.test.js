import { shallow } from "enzyme";
import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import Profile from "./Profile";

const mockStore = configureMockStore();
const store = mockStore({});
const mockProps = {
  match: {
    params: {
      id: 1
    }
  },
  getProfileByID: jest.fn(),
  profile: {
    profile: [
      {
        experience: [{ _id: 1 }],
        education: [{ _id: 2 }],
        githubusername: "john912",
        user: {
          _id: 1231
        }
      }
    ],
    loading: false
  },
  auth: {
    isAuthenticated: true,
    loading: false,
    user: {
      _id: 1231
    }
  }
};
const wrapper_filled = shallow(
  <Provider store={store}>
    <Profile {...mockProps} />
  </Provider>
);
describe("Test Profile Component", () => {
  it("renders component without crash", () => {
    expect(wrapper_filled.debug()).toMatchSnapshot();
  });
});
