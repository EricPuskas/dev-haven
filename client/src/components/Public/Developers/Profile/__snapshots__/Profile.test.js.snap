// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile Component renders component without crash 1`] = `
"<ContextProvider value={{...}}>
  <Connect(Profile) match={{...}} getProfileByID={[Function: mockConstructor]} profile={{...}} auth={{...}} />
</ContextProvider>"
`;
